internal/config/stub.go

```diff
@@ -1,80 +1,113 @@
-	return val, err
+func NewBlankConfig() *ConfigMock {
+	defaultStr := `
+# What protocol to use when performing git operations. Supported values: ssh, https
+git_protocol: https
+# What editor gh should run when creating issues, pull requests, etc. If blank, will refer to environment.
+editor:
+# When to interactively prompt. This is a global config that cannot be overridden by hostname. Supported values: enabled, disabled
+prompt: enabled
+# A pager program to send command output to, e.g. "less". Set the value to "cat" to disable the pager.
+pager:
+# Aliases allow you to create nicknames for gh commands
+aliases:
+  co: pr checkout
+# The path to a unix socket through which send HTTP connections. If blank, HTTP traffic will be handled by net/http.DefaultTransport.
+http_unix_socket:
+# What web browser gh should use when opening URLs. If blank, will refer to environment.
+browser:
+`
+	return NewFromString(defaultStr)
 }
 
-func (c ConfigStub) GetWithSource(host, key string) (string, string, error) {
-	if v, found := c[genKey(host, key)]; found {
-		return v, "(memory)", nil
+func NewFromString(cfgStr string) *ConfigMock {
+	c := ghConfig.ReadFromString(cfgStr)
+	cfg := cfg{c}
+	mock := &ConfigMock{}
+	mock.AuthTokenFunc = func(host string) (string, string) {
+		token, _ := c.Get([]string{"hosts", host, "oauth_token"})
+		return token, "oauth_token"
 	}
-	return "", "", errors.New("not found")
-}
-
-func (c ConfigStub) GetOrDefault(hostname, key string) (val string, err error) {
-	val, _, err = c.GetOrDefaultWithSource(hostname, key)
-	return
-}
-
-func (c ConfigStub) GetOrDefaultWithSource(hostname, key string) (val string, src string, err error) {
-	val, src, err = c.GetWithSource(hostname, key)
-	if err == nil && val == "" {
-		val = c.Default(key)
+	mock.GetFunc = func(host, key string) (string, error) {
+		return cfg.Get(host, key)
 	}
-	return
-}
-
-func (c ConfigStub) Default(key string) string {
-	return defaultFor(key)
-}
-
-func (c ConfigStub) Set(host, key, value string) error {
-	c[genKey(host, key)] = value
-	return nil
-}
-
-func (c ConfigStub) Aliases() (*AliasConfig, error) {
-	return nil, nil
-}
-
-func (c ConfigStub) Hosts() ([]string, error) {
-	return nil, nil
-}
-
-func (c ConfigStub) UnsetHost(hostname string) {
-}
-
-func (c ConfigStub) CheckWriteable(host, key string) error {
-	return nil
-}
-
-func (c ConfigStub) Write() error {
-	c["_written"] = "true"
-	return nil
-}
-
-func (c ConfigStub) WriteHosts() error {
-	return nil
-}
-
-func (c ConfigStub) DefaultHost() (string, error) {
-	return "", nil
+	mock.GetOrDefaultFunc = func(host, key string) (string, error) {
+		return cfg.GetOrDefault(host, key)
+	}
+	mock.SetFunc = func(host, key, value string) {
+		cfg.Set(host, key, value)
+	}
+	mock.UnsetHostFunc = func(host string) {
+		cfg.UnsetHost(host)
+	}
+	mock.HostsFunc = func() []string {
+		keys, _ := c.Keys([]string{"hosts"})
+		if host, ok := os.LookupEnv("GH_HOST"); ok && host != "" {
```

Comment:
I don't think that the mock config object should necessarily implement details like respecting GH_HOST. The mock config is to be used in place of a real config object; it doesn't necessarily need to implement the same semantics as the real config object. The GH_HOST behavior as it's implemented by the real config object is already tested within go-gh.